python/Ganga/GPIDev/Lib/File/OutputFileManager.py

```diff
@@ -210,53 +195,38 @@ def getWNCodeForDownloadingInputFiles(job, indent):
     """
 
     from Ganga.GPIDev.Lib.Dataset.GangaDataset import GangaDataset
-    if job.inputfiles is None or len(job.inputfiles) == 0 and\
-            (not job.inputdata or ((not isType(job.inputdata, GangaDataset)) or\
-                not job.inputdata.treat_as_inputfiles )):
-        return ""
+    if job.inputfiles is None or len(job.inputfiles) == 0 and not (job.inputdata or (isinstance(job.inputdata, GangaDataset) and job.inputdata.treat_as_inputfiles)):
+        if job.master is None or len(job.master.inputfiles) == 0 and not (job.master.inputdata or (isinstance(job.master.inputdata, GangaDataset) and job.master.inputdata.treat_as_inputfiles)):
```

Comment:
original test (not authored in this PR) is testing that the inputfiles are un-populated and that there is no inputdata which is to be treated as inputfiles for a job and then the test is repeated on the master job if one exists.

I would have argued the original form with breaklines made this easy to parse and that making this just 2 lines is what is causing the confusion here.